File: java-test/src/main/java/post/parthmistry/nst/capacity/SocketCountLogger.java
package post.parthmistry.nst.capacity;

import java.net.Socket;
import java.util.ArrayList;
import java.util.List;

public class SocketCountLogger extends Thread {

    private final List<Socket> sockets = new ArrayList<>();

    private final String name;

    public SocketCountLogger(String name) {
        this.name = name;
    }

    @Override
    public void run() {
        while (true) {
            try {
                Thread.sleep(2000);
                System.out.println(name + "Sockets: " + sockets.size());
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }

    public void addSocket(Socket socket) {
        sockets.add(socket);
    }

}

File: rust-test/src/socket_count_logger.rs
use std::net::TcpStream;
use std::sync::{Arc, Mutex};
use std::thread;
use std::thread::JoinHandle;
use std::time::Duration;

pub struct SocketCountLogger {
    sockets: Arc<Mutex<Vec<TcpStream>>>,
    pub join_handle: JoinHandle<()>
}

impl SocketCountLogger {

    pub fn new(name: &'static str) -> SocketCountLogger {
        let sockets = Arc::new(Mutex::new(Vec::new()));
        let sockets_clone = Arc::clone(&sockets);
        let join_handle = thread::spawn(move || {
            let sockets = sockets_clone;
            loop {
                thread::sleep(Duration::from_secs(2));
                println!("{}Sockets: {}", name, sockets.lock().unwrap().len());
            }
        });
        SocketCountLogger {
            sockets,
            join_handle
        }
    }

    pub fn add_socket(&mut self, socket: TcpStream) -> () {
        self.sockets.lock().unwrap().push(socket);
    }

}